lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -511,20 +598,92 @@ LogicalResult BuildOpGroups::buildOp(PatternRewriter &rewriter,
 
 LogicalResult BuildOpGroups::buildOp(PatternRewriter &rewriter,
                                      scf::YieldOp yieldOp) const {
-  if (yieldOp.getOperands().size() == 0)
+  if (yieldOp.getOperands().empty()) {
+    // If yiedOp operands are empty, we assume we have a for loop.
+    auto forOp = dyn_cast<scf::ForOp>(yieldOp->getParentOp());
+    assert(forOp);
+    ScfForOp forOpInterface(forOp);
+
+    // Get the ForLoop's Induction Register.
+    auto inductionReg =
+        getState<ComponentLoweringState>().getForLoopIterReg(forOpInterface, 0);
+
+    // Building the ``Latch'' Group for the For loop, which increments
+    // inductionReg by the step of the For loop. Gets an adder by mimicking what
+    // is dones in `buildLibraryOp`. There may be a more concise way to do this.
+    Type regWidth = inductionReg.getOut().getType();
+    // Adder should have same width as the inductionReg.
+    SmallVector<Type> types(3, regWidth);
+    auto addOp = getState<ComponentLoweringState>()
+                     .getNewLibraryOpInstance<calyx::AddLibOp>(
+                         rewriter, forOp.getLoc(), types);
+
+    auto directions = addOp.portDirections();
+    // For an add operation, we expect two input ports and one output port
+    SmallVector<Value, 2> opInputPorts;
+    Value opOutputPort;
+    for (auto dir : enumerate(directions)) {
+      if (dir.value() == calyx::Direction::Input)
```

Comment:
Prefer: `switch(dir.value()) { ... }` to ensure if any new directions are added, this code is updated.